cmd/openfga/main_test.go

```diff
@@ -203,6 +204,123 @@ func newOpenFGATester(t *testing.T, args ...string) OpenFGATester {
 	}
 }
 
+func TestCheckWithQueryCacheEnabled(t *testing.T) {
```

Comment:
can we rename this test to better explain what it is testing? e.g. `TestCheckPassesAlongContextualTuplesInDispatchedChecks`?

Reply:
Also curious why did we make this an integration test? It can be made a unit test that runs much faster

Reply:
Integration test is higher fidelity here, it was easier to write, and the same unit-test setup to exercise this test was proving to be awkward and more difficult as a unit test because of how we construct a LocalChecker per request.

Reply:
> can we rename this test to better explain what it is testing? e.g. TestCheckPassesAlongContextualTuplesInDispatchedChecks?

> Otherwise this makes it sound like the subject under test is the cache, when the problem is actually within Check

The problem is with Check, specifically the cached check resolver, and since the test was re-written using a test table pattern I felt it was reasonble to name the test TestCheckWithQueryCacheEnabled, because there will be other tests we add for testing query cache behavior I'm sure.. This isn't the only one that will ever exist.

Reply:
> The problem is with Check, specifically the cached check resolver,

To be more clear, by "check" I mean the LocalChecker. The cached check resolved made an assumption about LocalChecker that was not being maintained, this is why I say it's a problem within LocalChecker

Reply:
I could pass in a mocked delegate to the LocalChecker and assert that the contextual tuples are propagated. We already have a test that exercises the original reported issue with high fidelity though. If we feel that doing this is worth it in addition to what's been done then I will add another test for that as well in `interna/graph/check_test.go`.

Reply:
> proving to be awkward and more difficult as a unit test because of how we construct a LocalChecker per request.

This is part of why I sent https://github.com/openfga/openfga/pull/948 :) can you give it another look?

The next step to make it easier is to not have to do `context.WithTypesystem`.

Reply:
My concern with integ tests vs unit tests: i am trying to verify that commenting out each change breaks one test. And to do that, i need to comment all lines, run docker build, run the test. Uncomment a line, run docker build, run the test, etc.

Whereas with a unit test i don't need to run docker build. And I can put breakpoints easily.

Reply:
I don't disagree, but test fidelity and confidence is more important than convenience. We're testing to increase confidence not just for the sake of getting some rough coverage of changed code, and so I don't see anything wrong with favoring _functional tests_ in scenarios like this. If you're manually setting up the structures being tested, then you're not testing the system with as much fidelity as running the actual standalone server that end-users are going to be interacting with at the end of the day.

Reply:
> We're testing to increase confidence not just for the sake of getting some rough coverage of changed code

? Coverage gives you a metric on confidence. The only way i noted that our tests were not good enough was when i ran them after making commenting out the fixes.